api/client.go

```diff
@@ -389,10 +389,12 @@ func (c *Client) SetAddress(addr string) error {
 	defer c.modifyLock.Unlock()
 
 	var err error
-	if c.addr, err = url.Parse(addr); err != nil {
+	var parsedAddr *url.URL
+	if parsedAddr, err = url.Parse(addr); err != nil {
```

Comment:
Does it still make sense to declare the two vars above this line, instead of just like:
```
parsedAddr, err := url.Parse(addr)
if err != nil {
    ....
}
```

Reply:
Probably not, no. I'll change it.